ygen/enumgen.go

```diff
@@ -0,0 +1,461 @@
+package ygen
```

Comment:
Can you add the copyright and licence notice to the top of this file please? (It should be at the top of all code files across the repo.)